macros/schema_tests/multi-column/expect_compound_columns_to_be_unique.sql

```diff
@@ -28,19 +28,12 @@
     {{ row_condition }} and
 {% endif -%}
 
-{%- if ignore_row_if == "all_values_are_missing" %}
-        (
-            {% for column in columns -%}
-            {{ column }} is not null{% if not loop.last %} and {% endif %}
-            {% endfor %}
-        )
-{%- elif ignore_row_if == "any_value_is_missing" %}
-        (
-            {% for column in columns -%}
-            {{ column }} is not null{% if not loop.last %} or {% endif %}
-            {% endfor %}
-        )
-{%- endif -%}
+{%- set op = "and" if ignore_row_if == "all_values_are_missing" else "or" -%}
```

Comment:
This looks correct now, but closes off the "third behavior" of "never ignore" and somewhat obfuscates what behavior to expect if I pass a value other than `"all_values_are_missing"`. 

Did you mean to eliminate the (heretofore undocumented) third behavior?

What would you think of adding a guard prior to this that ensures the value of `ignore_row_if` is an expected one? Even if it's not documented, anybody reading the code would then know how to call it.

Reply:
> I also had the thought, is it possible that this logic shows up in other expectations? I didn't investigate this at all.
I'll take a look, good call out. If so, I'll open a separate issue/PR though.

Reply:
So, the built-in uniqueness check [always checks for nulls](https://github.com/dbt-labs/dbt-core/blob/23c8ac230ccf3e56413d3fa7d22fc62c5e51d7a3/core/dbt/include/global_project/macros/generic_test_sql/not_null.sql#L7) and excludes null values from the uniqueness check.
The default for the multi-column case (which is the generalized version of `expect_column_values_to_be_unique`) is to exclude rows where both columns are null from the uniqueness test, unless overridden.

I think I makes sense to check that the value of `ignore_row_if` is from a known set and error out otherwise.

Reply:
sounds good, i agree with this default behavior